lib/Config.js

```diff
@@ -1117,6 +1117,47 @@ class Config extends EventEmitter {
                 this.redis.password = config.redis.password;
             }
         }
+        if (config.scuba) {
+            if (config.scuba.host) {
+                assert(typeof config.scuba.host === 'string',
+                    'bad config: scuba host must be a string');
+                this.scuba = { host: config.scuba.host };
+            }
+            if (config.scuba.port) {
+                assert(Number.isInteger(config.scuba.port)
+                    && config.scuba.port > 0,
+                    'bad config: scuba port must be a positive integer');
+                this.scuba.port = config.scuba.port;
+            }
+        }
+        if (process.env.SCUBA_HOST && process.env.SCUBA_PORT) {
+            assert(typeof process.env.SCUBA_HOST === 'string',
+                'bad config: scuba host must be a string');
+            assert(Number.isInteger(Number(process.env.SCUBA_PORT))
+                && Number(process.env.SCUBA_PORT) > 0,
+                'bad config: scuba port must be a positive integer');
+            this.scuba = {
+                host: process.env.SCUBA_HOST,
+                port: Number(process.env.SCUBA_PORT),
+            };
+        }
+        if (this.scuba) {
+            this.quotaEnabled = true;
+        }
+        if (config.quota) {
+            if (config.quota.maxStalenessMS) {
+                assert(Number.isInteger(config.quota.maxStalenessMS),
+                    'bad config: quota.maxStalenessMS must be an integer');
+                this.quota = { maxStalenessMS: config.quota.maxStalenessMS };
+            }
+        } else {
+            this.quota = { maxStalenessMS: 24 * 60 * 60 * 1000 };
+        }
+        if (process.env.QUOTA_MAX_STALENESS_MS) {
+            assert(Number.isInteger(Number(process.env.QUOTA_MAX_STALENESS_MS)),
+                'bad config: quota.maxStalenessMS must be an integer');
```

Comment:
```suggestion
                'bad config: QUOTA_MAX_STALENESS_MS must be an integer');
```

Reply:
or better yet, merge this with the previous case, so we do this only once. something like:

```
const maxStaleness = process.env.QUOTA_MAX_STALENESS_MS || config.quota?.maxStatenessMS || 24 * 60 * 60 * 1000;
assert(Number.isInteger(maxStaleness), 'bad config: maxStalenessMS must be an integer');
this.quota = { maxStaleness };
```

Reply:
Updated